core-java-modules/core-java-string-algorithms-3/src/main/java/com/baeldung/emailvalidation/EmailValidation.java

```diff
@@ -0,0 +1,105 @@
+package com.baeldung.emailvalidation;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.validator.routines.EmailValidator;
+
+public class EmailValidation {
+
+    public static boolean usingEmailValidator(String emailAddress) {
+
+        EmailValidator validator = EmailValidator.getInstance();
+
+        return validator.isValid(emailAddress);
+    }
+
+    public static boolean usingSimpleRegEx(String emailAddress) {
+
+        String regExPattern = "^(.+)@(\\S+)$";
+
+        Pattern pattern = Pattern.compile(regExPattern);
+
+        Matcher matcher = pattern.matcher(emailAddress);
+
+        return matcher.matches();
+
+    }
+
+    public static boolean usingStrictRegEx(String emailAddress) {
+
+        String regExPattern = "^(?=.{1,64}@)[A-Za-z0-9_-]+(\\.[A-Za-z0-9_-]+)*@" 
+            + "[^-][A-Za-z0-9-]+(\\.[A-Za-z0-9-]+)*(\\.[A-Za-z]{2,})$";
+
+        Pattern pattern = Pattern.compile(regExPattern);
+
+        Matcher matcher = pattern.matcher(emailAddress);
+
+        return matcher.matches();
+
+    }
+
+    public static boolean usingUnicodeRegEx(String emailAddress) {
+
+        String regExPattern = "^(?=.{1,64}@)[\\p{L}0-9_-]+(\\.[\\p{L}0-9_-]+)*@" 
+            + "[^-][\\p{L}0-9-]+(\\.[\\p{L}0-9-]+)*(\\.[\\p{L}]{2,})$";
+
+        Pattern pattern = Pattern.compile(regExPattern);

```

Comment:
Can't we put the pattern code in a common private method to avoid duplicating it so many times?